lms/djangoapps/mobile_api/mobile_platform.py

```diff
@@ -0,0 +1,59 @@
+"""
+Platform related operations for Mobile APP
+"""
+import abc
+import re
+
+
+class MobilePlatform:
+    """
+    MobilePlatform class creates an instance of platform based on user agent and supports platform
+    related operations
+    """
+    __metaclass__ = abc.ABCMeta
+    version = None
+
+    def __init__(self, version):
+        self.version = version
+
+    @classmethod
+    def create_instance(cls, user_agent, user_agent_regex):
```

Comment:
This requires a better method name as it only conditionally returns an instance of this class.  So it's not a true factory method.  
Also, if this is meant to be an internal private method, it should be preceded with an underscore.